metadata/src/main/java/org/apache/kafka/controller/QuorumController.java

```diff
@@ -1065,6 +1065,11 @@ public void handleCommit(BatchReader<ApiMessageAndVersion> reader) {
                         int epoch = batch.epoch();
                         List<ApiMessageAndVersion> messages = batch.records();
 
+                        if (messages.isEmpty()) {
+                            log.debug("Skipping handling commit for batch with no data records with offset {} and epoch {}.", offset, epoch);
+                            continue;
+                        }
+    
```

Comment:
Let's get rid of this line which isn't needed:
```
// The active controller can delete up to the current committed offset.
snapshotRegistry.deleteSnapshotsUpTo(offsetControl.lastStableOffset());
```

since that is already done in `OffsetsControlManager.handleCommitBatch`.